test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -11007,22 +11007,24 @@ def add_server_dyn_res(self, custom_resource, script_body=None,
                          configuration settings.
         :type validate: bool
         """
+        file_specified = False
         if res_file is not None:
             f = open(file)
             script_body = f.readlines()
             f.close()
+            file_specified = True
         else:
-            res_file = self.du.create_temp_file(prefix='PtlPbsSchedConfig',
-                                                body=script_body,
-                                                asuser="root")
+            res_file = self.du.create_dyn_res_script(script_body,
+                                                     prefix='\
+                                                     PtlPbsSchedConfig')
 
         self.server_dyn_res = res_file
         self.logger.info(self.logprefix + "adding server dyn res " + res_file)
         self.logger.info("-" * 30)
         self.logger.info(script_body)
         self.logger.info("-" * 30)
-
-        self.du.chmod(self.hostname, path=res_file, mode=0755)
+        if file_specified is True:
```

Comment:
you can just say "if file_specified"

Reply:
also, why not just do the chmod as part of the "if res_file is not None" block?

Reply:
You are right, it was silly of me to not do that instead

Reply:
Ah, ok, i guess it's just personal preference then

Reply:
I take it back.  I checked the pep-8 style guide and they say:
Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:

I guess I'll be changing how I code with booleans now.